docs: configure Rslib (#352)
web-infra-dev/rslib: website/docs/en/guide/basic/configure-rslib.mdx

      },
      // Independent Rsbuild configuration
      output: {
        // options for build outputs
      },
      source: {
        // options for source code parsing and compilation
      },
      tools: {
        // options for the low-level tools
      },
      plugins: [
        // configure Rsbuild plugins
      ],
      // other independent Rsbuild configuration options
    },
    {
      // The second set of configurations of CJS format
      format: 'cjs',
      // ... other independent Rslib / Rsbuild configurations
    },
    // ... Extra sets of configurations (if needed)
  ],
  // Shared Rsbuild configuration which will be merged into each of the above lib configuration objects
  output: {
    // options for build outputs
  },
  source: {
    // options for source code parsing and compilation
  },
  tools: {
    // options for the low-level tools
  },
  plugins: [
    // configure Rsbuild plugins
  ],
  // ... other Rsbuild configuration options
};
```

The configuration file exports a default object that includes a `lib` array and shared Rsbuild configuration options. You can find detailed descriptions of all configs on the [Configure Overview](/config/) page.

> - **Lib Configuration (required)**: The `lib` array contains multiple objects, each representing a set of independent configurations tailored for different output formats, such as ESM and CJS. Each object within the `lib` array can specify distinct configurations for both Rslib and Rsbuild.

Reviewer: Mentioning both Rslib and Rsbuild here might be a bit confused for users. Rslib's config extends Rsbuild's, so we could tell user you can specify distict Rslib config here. And then tell them Rslib is a superset of Rsbuild in `lib`.

>   - **Independent Rslib Configuration (required)**: The configuration options provided by Rslib. Rslib will transform these options into complex Rsbuild configuration of each object.

Reviewer: Some of the configs are not required, only `format` is required.

>   - **Independent Rslib Configuration (required)**: The configuration options provided by Rslib. Rslib will transform these options into complex Rsbuild configuration of each object.

Reviewer: ```suggestion
  - **Independent Rslib Configuration (required)**: The configuration options provided by Rslib. Rslib will transform these options into adapted Rsbuild configuration of each object.
```

          ? './src/request.dev.js'
          : './src/request.prod.js',
    },
  },
});
```

## Configure Rsbuild

In Rslib project, you can configure Rsbuild related configurations in both `lib` configuration object and shared configuration section.

Common-used Rsbuild configurations in library development are listed at [Rsbuild Configuration](/config/rsbuild).

You can also refer to the [Rsbuild Configuration](https://rsbuild.dev/config/index#config-overview) for checking the complete configuration items.

## Configure Rspack

Rslib is built on top of Rsbuild and Rsbuild supports directly modifying the Rspack configuration object and also supports modifying the built-in Rspack configuration of Rsbuild through `rspack-chain`. This means you can configure Rspack related configurations in an Rslib project as well.

For more details, refer to [Configure Rspack](https://rsbuild.dev/guide/basic/configure-rspack).

## Configuration Debug

You can enable Rslib's debug mode by adding the `DEBUG=rsbuild` environment variable when executing a build. It will display the final Rsbuild/Rspack configuration after processing by Rslib.

```bash
DEBUG=rsbuild pnpm build
```

> In debug mode, Rslib will write the Rsbuild / Rspack config to the dist directory, which is convenient for developers to view and debug.

Reviewer: ```suggestion
In debug mode, Rslib will write the Rsbuild / Rspack config to the dist directory, which is convenient for developers to view and debug. Here’s an example of a configuration for Rslib that builds both CJS and ESM output.
```

```

> success Inspect config succeed, open following files to view the content:

Reviewer: ```suggestion
Inspect config succeed, open following files to view the content:
```

https://github.com/web-infra-dev/rsbuild/blob/main/website/docs/en/guide/basic/cli.mdx?plain=1#L172

      output: {
        // options for build outputs
      },
      source: {
        // options for source code parsing and compilation
      },
      tools: {
        // options for the low-level tools
      },
      plugins: [
        // configure Rsbuild plugins
      ],
      // other independent Rsbuild configuration options
    },
    {
      // The second set of configurations of CJS format
      format: 'cjs',
      // ... other independent lib configurations and Rsbuild configurations
    },
    // ... Extra sets of configurations (if needed)
  ],
  // Shared Rsbuild configuration which will be merged into each of the above lib configuration objects
  output: {
    // options for build outputs
  },
  source: {
    // options for source code parsing and compilation
  },
  tools: {
    // options for the low-level tools
  },
  plugins: [
    // configure Rsbuild plugins
  ],
  // ... other Rsbuild configuration options
};
```

The configuration file exports a default object that includes a `lib` array and shared Rsbuild configuration options. You can find detailed descriptions of all configs on the [Configure Overview](/config/) page.

> - **[Lib Configuration](/config/lib) (at least one object is required)**: The `lib` array contains multiple objects, each representing a set of independent configurations tailored for different output formats, such as ESM and CJS. Each object within the `lib` array can specify distinct configurations, which is a superset of Rsbuild configurations.

Reviewer: ```suggestion
- **[Lib Configuration](/config/lib) (at least one object is required)**: The `lib` array contains multiple objects, each representing a set of independent configurations and will generate different outputs. Each object within the `lib` array can specify distinct configurations, which is a superset of Rsbuild configurations, with Rslib's unique configuration.

The configuration options extended by Rslib will be transformed into adapted Rsbuild configuration of each object. The configuration options provided by Rsbuild which can override the shared Rsbuild configuration outside.
```

>   - **Independent [lib Configuration](/config/lib) (only [format](/config/lib/format) is required)**: The configuration options extended by Rslib. Rslib will transform these options into adapted Rsbuild configuration of each object.

Reviewer: ```suggestion
```

>   - **Independent [Rsbuild Configuration](/config/rsbuild) (optional)**: The configuration options provided by Rsbuild which can override the shared Rsbuild configuration outside.

Reviewer: ```suggestion
```
